src/cljs/ataru/hakija/hakija_readonly.cljs

```diff
@@ -18,40 +18,60 @@
                                                                        question-group-answer?
-                                                                       answers->read-only-format]]
+                                                                       answers->read-only-format
+                                                                       value-or-koodi-uri->label
+                                                                       group-spacer]]
             [taoensso.timbre :refer-macros [spy debug]]))
 
 (defn- multiple-choice-with-koodisto [field-descriptor]
   (and (= (:fieldType field-descriptor) "multipleChoice")
        (contains? field-descriptor :koodisto-source)))
 
-(defn text [field-descriptor application lang]
+(defn- multi-values->:li
+  [field-descriptor lang values]
+  (map-indexed (fn [idx value-or-koodi-uri]
+                 (let [value (str (if (map? value-or-koodi-uri)
+                                    (:value value-or-koodi-uri)
+                                    value-or-koodi-uri))]
+                   ^{:key (str "value-" idx)}
+                   [:li (value-or-koodi-uri->label field-descriptor lang value)]))
+               values))
+
+(defn text [field-descriptor application lang question-group-index]
   (let [answer            ((answer-key field-descriptor) (:answers application))
-        multi-values->:li (partial map-indexed (fn [idx value]
-                                                 ^{:key (str "value-" idx)}
-                                                 [:li (str (if (map? value)
-                                                             (:value value)
-                                                             value))]))]
+        values (if question-group-index
+                 (-> answer :values (nth question-group-index))
+                 (:values answer))]
     [:div.application__form-field
      [:label.application__form-field-label
       (str (-> field-descriptor :label lang) (required-hint field-descriptor))]
      (if (:cannot-view answer)
        [:div "***********"]
-       [:div
-        (cond (and (vector? (:values answer))
-                   (every? vector? (:values answer)))
-              (into [:ul.application__form-field-list] (map multi-values->:li (:values answer)))
-
-              (and (vector? (:values answer))
-                   (some (comp not vector?) (:values answer)))
-              (into [:ul.application__form-field-list] (multi-values->:li (:values answer)))
+       [:div.application__readonly-text
+        (cond (and (vector? values)
+                   (every? vector? values)
+                   (not= 1 (count values)))
+              (into [:ul.application__form-field-list]
+                    (map #(multi-values->:li field-descriptor lang %) values))
+
+              (and (vector? values)
+                   (some (comp not vector?) values)
```

Comment:
I'm thinking there might be a possibility this does return false positives on some cases. Please see my previous comment above.